### PR TITLE
Notest draingen

### PR DESCRIPTION
**Pull request by akhanzode**

### CODE REVIEW

File: src/frontend/org/voltdb/export/PushSpecificGeneration.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                                    = "Custom Export failed to configure, failed to load"
+                                    + " export plugin class: " + exportConfiguration.getExportconnectorclass()
+                                    + " Disabling export.";
+                            throw new CatalogUtil.DeploymentCheckException(msg);
+                        }
+                    }
+                    break;
+            }
+            StandaloneExportManager.initialize(0, args[1], exportClientClassName, dep.getExport().getConfiguration().get(0).getProperty());
+            int maxPart = dep.getCluster().getSitesperhost();
```
Comment (vtkstef) [inline src/frontend/org/voltdb/export/PushSpecificGeneration.java]: is the scope the cluster's or the host? K-Factor?

File: src/frontend/org/voltdb/export/StandaloneExportGeneration.java [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    private int m_numSources = 0;
+    private final AtomicInteger m_drainedSources = new AtomicInteger(0);
+
+    private Runnable m_onAllSourcesDrained = null;
+
+    private final Runnable m_onSourceDrained = new Runnable() {
+        @Override
+        public void run() {
+            if (m_onAllSourcesDrained == null) {
+                VoltDB.crashLocalVoltDB("No export generation roller found.", true, null);
```
Comment (vtkstef) [inline src/frontend/org/voltdb/export/StandaloneExportGeneration.java]: crash ...

File: src/frontend/org/voltdb/export/StandaloneExportGeneration.java [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+                        break;
+                    }
+                }
+
+                if (haveDataFiles) {
+                    try {
+                        addDataSource(f, partitions);
+                        hadValidAd = true;
+                    } catch (IOException e) {
+                        VoltDB.crashLocalVoltDB("Error intializing export datasource " + f, true, e);
```
Comment (vtkstef) [inline src/frontend/org/voltdb/export/StandaloneExportGeneration.java]: crash ...

File: src/frontend/org/voltdb/export/StandaloneExportGeneration.java [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+     */
+    public StandaloneExportGeneration(File generationDirectory) throws IOException {
+        m_directory = generationDirectory;
+    }
+
+    public boolean isContinueingGeneration() {
+        return false;
+    }
+
+    boolean initializeGenerationFromDisk(final Connector conn, HostMessenger messenger) {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/export/StandaloneExportGeneration.java]: maybe rename messenger parameter to ignored

File: src/frontend/org/voltdb/export/StandaloneExportManager.java [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+    }
+
+    /*
+     * This method pulls double duty as a means of pushing export buffers
+     * and "syncing" export data to disk. Syncing doesn't imply fsync, it just means
+     * writing the data to a file instead of keeping it all in memory.
+     * End of stream indicates that no more data is coming from this source
+     * for this generation.
+     */
+    public static void pushExportBuffer(
```
Comment (vtkstef) [inline src/frontend/org/voltdb/export/StandaloneExportManager.java]: may remove

File: src/frontend/org/voltdb/export/PushSpecificGeneration.java (hunk 1 above)
Comment (akhanzode) [inline src/frontend/org/voltdb/export/PushSpecificGeneration.java]: This is regardless of mastership etc.

File: src/frontend/org/voltdb/export/StandaloneExportGeneration.java (hunk 2 above)
Comment (akhanzode) [inline src/frontend/org/voltdb/export/StandaloneExportGeneration.java]: It should be just an exit. I can change this to just that if you like with printing of message.

File: src/frontend/org/voltdb/export/StandaloneExportGeneration.java (hunk 4 above)
Comment (akhanzode) [inline src/frontend/org/voltdb/export/StandaloneExportGeneration.java]: sure

**Comment by akhanzode** [comment]
squash merged

**Pull request closed without merging**